File: TheaterSuggestions/Java/SeatsSuggestions/src/main/java/com/octo/technology/SeatsSuggestions/SeatingOptionSuggested.java
package com.octo.technology.SeatsSuggestions;

import java.util.ArrayList;
import java.util.List;

public class SeatingOptionSuggested {

    private final List<Seat> seats = new ArrayList<>();
    private final int partyRequested;

    public SeatingOptionSuggested(SuggestionRequest suggestionRequest) {
        this.partyRequested = suggestionRequest.partyRequested();
    }

    public void addSeat(Seat seat) {
        seats.add(seat);
    }

    public boolean matchExpectation() {
        return seats.size() == partyRequested;
    }

    public List<Seat> seats() {
        return seats;
    }
}

File: TheaterSuggestions/CSharp/SeatsSuggestions/SeatingOptionSuggested.cs
using System.Collections.Generic;

namespace SeatsSuggestions
{
    public class SeatingOptionSuggested(int partyRequested, PricingCategory priceCategory)
    {
        public PricingCategory PricingCategory { get; } = priceCategory;
        public List<Seat> Seats { get; } = new();
        public int PartyRequested { get; } = partyRequested;

        public void AddSeat(Seat seat)
        {
            Seats.Add(seat);
        }

        public bool MatchExpectation()
        {
            return Seats.Count == PartyRequested;
        }
    }
}